build-conflicts.php

```diff
@@ -131,37 +129,12 @@ static function ($errorCode, $message = '', $file = '', $line = 0) : void {
         ));
     };
 
-    /** @return Advisory[] */
-    $findAdvisories = static function (string $path) use ($advisoriesExtension) : array {
-        return array_map(
-            static function (SplFileInfo $advisoryFile) {
-                $filePath = $advisoryFile->getRealPath();
-
-                assert(is_string($filePath));
-
-                return Advisory::fromArrayData(
-                    Yaml::parse(file_get_contents($filePath), Yaml::PARSE_EXCEPTION_ON_INVALID_TYPE)
-                );
-            },
-            iterator_to_array(new CallbackFilterIterator(
-                new RecursiveIteratorIterator(
-                    new RecursiveDirectoryIterator($path, FilesystemIterator::SKIP_DOTS),
-                    RecursiveIteratorIterator::SELF_FIRST
-                ),
-                static function (SplFileInfo $advisoryFile) use ($advisoriesExtension) {
-                    // @todo skip `vendor` dir
-                    return $advisoryFile->isFile() && $advisoryFile->getExtension() === $advisoriesExtension;
-                }
-            ))
-        );
-    };
-
     /**
-     * @param Advisory[] $advisories
+     * @param Generator $getAdvisories
```

Comment:
`Generator<Advisory>`

Reply:
They are templated types (usually called "generics"): I don't think PSR-5 covers them, but there's some docs for:

 * psalm: https://psalm.dev/docs/annotating_code/templated_annotations/
 * phpstan: https://arnaud.le-blanc.net/post/phpstan-generics.html

Reply:
Also: no shame in asking. This is a bi-directional process: I'm learning about github's security API, you are learning about some stricter PHP coding styles :+1: